tests/common/connections/ssh_console_conn.py

```diff
@@ -26,7 +27,13 @@ def __init__(self, **kwargs):
         super(SSHConsoleConn, self).__init__(**kwargs)
 
     def session_preparation(self):
-        self._test_channel_read()
+        session_init_msg = self._test_channel_read()
+        self.logger.debug(session_init_msg)
+
+        if re.search(r"Port is in use. Closing connection...", session_init_msg, flags=re.M):
+            console_port = self.username.split(':')[-1]
+            raise PortInUseException(f"Host closed connection, as console port '{console_port}' is currently occupied.")
```

Comment:
@matthew-soulsby what's the result for case dut_console/test_idle_timeout if this exception happens? Will it capture the exception and give a more specific failed reason with pytest.fail?

Reply:
@ZhaohuiS So the exception is left uncaptured, and the way that it is displayed is as follows (extracted from the test run xml):
```xml
<error message="failed on setup with `tests.common.connections.ssh_console_conn.PortInUseException: Host aborted connection, as console port '36' is currently occupied.`">
```

All of this executes in the setup [here](https://github.com/sonic-net/sonic-mgmt/blob/9e3cc9b8394adf8ce9fa2215adec4f17d0cb7e27/tests/conftest.py#L1733), prior to the test case itself being run, so I don't believe that pytest.fail would work here, but the exception thrown provides useful information (regarding the device, port and the port being in use), along with the fact that it was the setup that failed.

Happy to make changes if there's a better way of doing it though!